tests/check/check.go

```diff
@@ -196,3 +199,430 @@ func runTest(t *testing.T, test individualTest, params testParams, contextTupleT
+					Expectation: false,
+				},
+				{
+					Name:        "invalid_cond_not_public_id",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:not-public", Relation: "direct_wild_cond", User: "user:valid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("2")}},
+					Expectation: false,
+				},
+			},
+		},
+	},
+}
+
+func runTestMatrix(t *testing.T, params testParams) {
+	schemaVersion := params.schemaVersion
+	client := params.client
+	name := matrix.Name
+
+	ctx := context.Background()
+	resp, err := client.CreateStore(ctx, &openfgav1.CreateStoreRequest{Name: name})
+	require.NoError(t, err)
+	storeID := resp.GetId()
+
+	var modelID string
+	t.Run(name, func(t *testing.T) {
+		for _, stage := range matrix.Stages {
+			t.Run(fmt.Sprintf("stage_%s", stage.Name), func(t *testing.T) {
+				if stage.Model != "" {
+					model := testutils.MustTransformDSLToProtoWithID(stage.Model)
+					writeModelResponse, err := client.WriteAuthorizationModel(ctx, &openfgav1.WriteAuthorizationModelRequest{
+						StoreId:         storeID,
+						SchemaVersion:   schemaVersion,
+						TypeDefinitions: model.GetTypeDefinitions(),
+						Conditions:      model.GetConditions(),
+					})
+					require.NoError(t, err)
+					modelID = writeModelResponse.GetAuthorizationModelId()
+				}
+
+				tuples := stage.Tuples
+				tuplesLength := len(tuples)
+				if tuplesLength > 0 {
+					for i := 0; i < tuplesLength; i += writeMaxChunkSize {
+						end := int(math.Min(float64(i+writeMaxChunkSize), float64(tuplesLength)))
+						writeChunk := (tuples)[i:end]
+						_, err = client.Write(ctx, &openfgav1.WriteRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: modelID,
+							Writes: &openfgav1.WriteRequestWrites{
+								TupleKeys: writeChunk,
+							},
+						})
+						require.NoError(t, err)
+					}
+				}
+
+				if len(stage.CheckAssertions) == 0 {
+					t.Skipf("no check assertions defined")
+				}
+				for _, assertion := range stage.CheckAssertions {
+					t.Run(fmt.Sprintf("assertion_%s", assertion.Name), func(t *testing.T) {
+						detailedInfo := fmt.Sprintf("Check request: %s. Tuples: %s. Contextual tuples: %s", assertion.Tuple, stage.Tuples, assertion.ContextualTuples)
+
+						var tupleKey *openfgav1.CheckRequestTupleKey
+						if assertion.Tuple != nil {
+							tupleKey = &openfgav1.CheckRequestTupleKey{
+								User:     assertion.Tuple.GetUser(),
+								Relation: assertion.Tuple.GetRelation(),
+								Object:   assertion.Tuple.GetObject(),
+							}
+						}
+						resp, err := client.Check(ctx, &openfgav1.CheckRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: modelID,
+							TupleKey:             tupleKey,
+							ContextualTuples: &openfgav1.ContextualTupleKeys{
+								// TODO
+								TupleKeys: []*openfgav1.TupleKey{},
+							},
+							Context: assertion.Context,
+							Trace:   true,
```

Comment:
Do we want trace to be true or false?  (Most people run trace off ... do we want our test to mimic that?)

Reply:
Turning off.